code/modules/client/preferences/names.dm

```diff
@@ -42,6 +42,7 @@
 /datum/preference/name/real_name/apply_to_human(mob/living/carbon/human/target, value)
 	target.real_name = value
 	target.name = value
+	log_mob_tag("\[[target.tag]\] RENAMED: [key_name(target)]")
```

Comment:
The real_name preference is applied every time a human is initialized. Are you sure this isn't duplicating any logs?

Reply:
This is also going to log for every single preference dummy which is not going to be useful

Reply:
Well if you for example look at https://tgstation13.org/parsed-logs/terry/data/logs/2022/10/01/round-191559/mob_tags.txt and check mob_3242 you will see these two lines:
TAG: [mob_3242] CREATED: *no key*/(Brooke Pearsall)
TAG: [mob_3242] NEW OWNER: MrBaguette/(Caden Gronko)
So clearly the name change of the players pref being applied isn't getting logged
Maybe real_name/apply_to_human should call fully_replace_character_name instead?
The dummys getting getting logged is indeed quite annoying, especially since the name change happens twice when opening the character settings.
Its already a problem since for example all the dummy creations are all getting logged in /mob/Initialize.
I feel like adding some if clause that checks if the initialized/renamed mob is a dummy and not logging it is kinda too snowflaky, got any better ideas how to exclude them?

Reply:
Have `log_mob_tag` be a proc on `/mob` that automatically adds `src.tag` and on dummies override it to be a no-op.

Reply:
Thats actually great way of dealing with it, commited the changes and also found out that mob/dead creation wasn't getting logged
Joining into a round now generates the following log entries:
TAG: [mob_2916] SPECIES: *no key*/(Ruben Wise) [/datum/species/human]
TAG: [mob_2916] CREATED: *no key*/(Ruben Wise) [/mob/living/carbon/human]
TAG: [mob_2916] SPECIES: *no key*/(Ruben Wise) [/datum/species/lizard]
TAG: [mob_2916] RENAMED: *no key*/(Taylor Mull)
TAG: [mob_2916] NEW OWNER: Guest-511315173/(Taylor Mull)
I sadly don't think there is an easy/clean way of preventing the first SPECIES log which is generated by the new human getting created for the player and then later applying the prefs to that

Reply:
That's fine